stl/inc/xutility

```diff
@@ -1912,6 +1857,135 @@ namespace ranges {
+    constexpr decltype(auto) _Unwrap_sent(_Sent&& _Se) noexcept(
+        !_Unwrappable_sentinel_for<_Sent, _Iter> || _Has_nothrow_unwrapped<_Sent>)
+    {
+        _STL_INTERNAL_STATIC_ASSERT(sentinel_for<remove_cvref_t<_Sent>, remove_cvref_t<_Iter>>);
+        if constexpr (is_pointer_v<remove_cvref_t<_Sent>>) {
+            return _Se + 0;
+        } else if constexpr (_Unwrappable_sentinel_for<_Sent, _Iter>) {
+            return static_cast<_Sent&&>(_Se)._Unwrapped();
+        } else {
+            return static_cast<_Sent&&>(_Se);
+        }
+    }
+
+    template <range _Rng, class _Iter>
+    constexpr decltype(auto) _Unwrap_range_iter(_Iter&& _It)
+        noexcept(noexcept(_Unwrap_iter<sentinel_t<_Rng>>(static_cast<_Iter&&>(_It))))
+    {
+        _STL_INTERNAL_STATIC_ASSERT(same_as<remove_cvref_t<_Iter>, iterator_t<_Rng>>);
+        return _Unwrap_iter<sentinel_t<_Rng>>(static_cast<_Iter&&>(_It));
+    }
+
+    template <range _Rng, class _Sent>
+    constexpr decltype(auto) _Unwrap_range_sent(_Sent&& _Se)
+        noexcept(noexcept(_Unwrap_sent<iterator_t<_Rng>>(static_cast<_Sent&&>(_Se))))
+    {
+        _STL_INTERNAL_STATIC_ASSERT(same_as<remove_cvref_t<_Sent>, sentinel_t<_Rng>>);
+        return _Unwrap_sent<iterator_t<_Rng>>(static_cast<_Sent&&>(_Se));
+    }
+
+    template <class _Iter, class _Sent>
+    using _Unwrap_iter_t = remove_cvref_t<decltype(_Unwrap_iter<_Sent>(_STD declval<_Iter>()))>;
+    template <class _Sent, class _Iter>
+    using _Unwrap_sent_t = remove_cvref_t<decltype(_Unwrap_sent<_Iter>(_STD declval<_Sent>()))>;
+
+    template <range _Rng>
+    using _Unwrapped_iterator_t = _Unwrap_iter_t<iterator_t<_Rng>, sentinel_t<_Rng>>;
+    template <range _Rng>
+    using _Unwrapped_sentinel_t = _Unwrap_sent_t<sentinel_t<_Rng>, iterator_t<_Rng>>;
+
+    namespace _Unchecked_begin {
+        template <class _Ty>
+        concept _Has_member = requires(_Ty& __t) {
+            { __t._Unchecked_begin() } -> input_or_output_iterator;
+        };
+
+        template <class _Ty>
+        concept _Can_begin = requires(_Ty& __t) {
+            _Unwrap_range_iter<_Ty>(_RANGES begin(__t));
+        };
+
+        class _Cpo {
+        private:
+            enum class _St { _None, _Member, _Unwrap };
+
+            template <class _Ty>
+            _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                _STL_INTERNAL_STATIC_ASSERT(is_lvalue_reference_v<_Ty>);
+                if constexpr (_Has_member<_Ty>) {
+                    _STL_INTERNAL_STATIC_ASSERT(
+                        same_as<decltype(_STD declval<_Ty>()._Unchecked_begin()), _Unwrapped_iterator_t<_Ty>>);
+                    return {_St::_Member, noexcept(_Fake_copy_init(_STD declval<_Ty>()._Unchecked_begin()))};
+                } else if constexpr (_Can_begin<_Ty>) {
+                    return {_St::_Unwrap, noexcept(_Fake_copy_init(_Unwrap_range_iter<_Ty>(_RANGES begin(_STD declval<_Ty>()))))};

```

Comment:
This line exceeds 120 columns, causing code format validation to fail. Please double-check that format-on-save is enabled in your editor (clang-format appears to be enabled for this region, it's disabled on line 1967 below).

Reply:
clang-format was disabled for this region accidentally (line 1879 doesn't have a matching clang-format on)